Examples refactor (#329)
ollama/ollama-python: README.md

The Ollama Python library provides the easiest way to integrate Python 3.8+ projects with [Ollama](https://github.com/ollama/ollama).

## Prerequisites

- [Ollama](https://ollama.com/download) should be installed and running

> - Pull a model to use with the library: `ollama pull <model>` e.g. `ollama pull llama3.1`

Reviewer: llama3.2 should be used for all examples where appropriate

```suggestion
- Pull a model to use with the library: `ollama pull <model>` e.g. `ollama pull llama3.2`
```

>   - See [Ollama models](https://ollama.com/models) for more information on the models available.

Reviewer: ```suggestion
  - See [Ollama.com](https://ollama.com/search) for more information on the models available.
```

## Usage

```python

> from ollama import chat
> from ollama._types import ChatResponse

Reviewer: `ollama._types` should not be imported directly. `ChatResponse` is exported explicitly by `ollama`. Also why the change to `from ollama import ...`?

```suggestion
from ollama import chat, ChatResponse
```

Author: Oh you're right let me fix

## Usage

```python
from ollama import chat
from ollama import ChatResponse

response: ChatResponse = chat(model='llama3.2', messages=[
  {
    'role': 'user',
    'content': 'Why is the sky blue?',
  },
])
print(response['message']['content'])
# or access fields directly from the response object
print(response.message.content)
```

See [_types.py](ollama/_types.py) for more information on the response types.

## Streaming responses

> Response streaming can be enabled by setting `stream=True`

Reviewer: ```suggestion
Response streaming can be enabled by setting `stream=True`.
```

> Streaming Tool/Function calling is not yet supported.

Reviewer: You can call out things in GitHub markdown with `> [!TAG]`

```suggestion
> [!NOTE]
> Streaming Tool/Function calling is not yet supported.
```

  print(chunk['message']['content'], end='', flush=True)
```

## Custom client

> A custom client can be created with the fields from the [httpx client](https://www.python-httpx.org/api/#client).

Reviewer: This reads weird. Focus on a custom Ollama client until `extra arguments are passed to httpx.Client` at the end

Author: Yeah reading this again it doesn't read well- will workshop

  print(chunk['message']['content'], end='', flush=True)
```

## Custom client

A custom client can be created with the following fields:

- `host`: The Ollama host (default: `http://localhost:11434`)
- `timeout`: The timeout for requests (default: `None`)
- `follow_redirects`: Whether to follow redirects (default: `True`)

> - `headers`: Additional headers to send with requests (default: `{}`)

Reviewer: The user is not restricted to these fields. Any extra kwargs are passed verbatim to `httpx.Client`. These are mentioned only because they're default values differ from `httpx`

Author: Okay - let me make that callout instead

```python
from ollama import Client
# default client
client = Client()

> # or custom client

Reviewer: This section is on custom clients so mentioning the default client seems unnecessary

  print(chunk['message']['content'], end='', flush=True)
```

## Custom client
A custom client can be created by instantiating `Client` or `AsyncClient` from `ollama`.

> All extra kwargs are passed into the [httpx client](https://www.python-httpx.org/api/#client).

Reviewer: nit:

```suggestion
All extra keyword arguments are passed into the [`httpx.Client`](https://www.python-httpx.org/api/#client).
```
